projection/src/main/scala/akka/projection/dynamodb/javadsl/DynamoDBProjection.scala

```diff
@@ -72,4 +75,52 @@ object DynamoDBProjection {
       .asInstanceOf[ExactlyOnceProjection[Offset, Envelope]]
   }
 
+  /**
+   * Create a [[akka.projection.Projection]] that groups envelopes and calls the `handler` with a group of `Envelopes`.
+   * The envelopes are grouped within a time window, or limited by a number of envelopes, whatever happens first. This
+   * window can be defined with [[GroupedProjection.withGroup]] of the returned `GroupedProjection`. The default
+   * settings for the window is defined in configuration section `akka.projection.grouped`.
+   *
+   * The offset is stored in DynamoDB in the same transaction as the `TransactWriteItem`s returned by the `handler`.
+   */
+  def exactlyOnceGroupedWithin[Offset, Envelope](
+      projectionId: ProjectionId,
+      settings: Optional[DynamoDBProjectionSettings],
+      sourceProvider: SourceProvider[Offset, Envelope],
+      handler: Supplier[DynamoDBTransactHandler[java.util.List[Envelope]]],
+      system: ActorSystem[_]): GroupedProjection[Offset, Envelope] = {
+    scaladsl.DynamoDBProjection
+      .exactlyOnceGroupedWithin[Offset, Envelope](
+        projectionId,
+        settings.asScala,
+        JavaToScalaBySliceSourceProviderAdapter(sourceProvider),
+        () => new DynamoDBTransactGroupedHandlerAdapter(handler.get()))(system)
+      .asInstanceOf[GroupedProjection[Offset, Envelope]]
+  }
+
+  /**
+   * Create a [[akka.projection.Projection]] that groups envelopes and calls the `handler` with a group of `Envelopes`.
+   * The envelopes are grouped within a time window, or limited by a number of envelopes, whatever happens first. This
+   * window can be defined with [[GroupedProjection.withGroup]] of the returned `GroupedProjection`. The default
+   * settings for the window is defined in configuration section `akka.projection.grouped`.
+   *
+   * It stores the offset in a relational database table using R2DBC immediately after the `handler` has processed the
+   * envelopes, but that is still with at-least-once processing semantics. This means that if the projection is
+   * restarted from previously stored offset the previous group of envelopes may be processed more than once.
```

Comment:
javadsl still referencing r2dbc, copying over from the scaladsl doc comment:

```suggestion
   * The offset is stored in DynamoDB immediately after the `handler` has processed the envelopes, but that is still
   * with at-least-once processing semantics. This means that if the projection is restarted from previously stored
   * offset the previous group of envelopes may be processed more than once.
```